plugin/testing_write_upsert.go

```diff
@@ -132,3 +132,74 @@ func (s *WriterTestSuite) testUpsertAll(ctx context.Context) error {
 
 	return nil
 }
+
+func (s *WriterTestSuite) testInsertDuplicatePK(ctx context.Context) error {
```

Comment:
Would it make sense to build the table and data manually instead of using `schema.TestTable` and `tg.Generate`?
This seems like a specific use case so not sure we need to expose another option to the test generator (also we override the ID column, so might as well build the table ourselves). See reference in `testUpsertBasic`

I think that would also make it easier to do the validation in the test, as we mostly want to validate the insert didn't fail and the most recent row was persisted